packages/subiquity_client/lib/src/types.dart

```diff
@@ -71,23 +101,63 @@ class ApplicationStatus with _$ApplicationStatus {
       _$ApplicationStatusFromJson(json);
 }
 
-Map<int, String> _keycodesFromJson(List<dynamic> keycodes) {
-  return {
-    for (final keycode in keycodes) keycode[0] as int: keycode[1] as String,
-  };
+enum PasswordKind {
+  NONE,
+  KNOWN,
+  UNKNOWN,
+}
+
+@freezed
+class KeyFingerprint with _$KeyFingerprint {
+  const factory KeyFingerprint({
+    required String keytype,
+    required String fingerprint,
+  }) = _KeyFingerprint;
+
+  factory KeyFingerprint.fromJson(Map<String, dynamic> json) =>
+      _$KeyFingerprintFromJson(json);
 }
```

Comment:
there are a few new types that aren't in use in the GUI today that codecov complains about. the total coverage goes only down by 0.7% so i think it's not a big deal :)